glean_parser/util.py

```diff
@@ -66,13 +67,33 @@ def remove_implicit_resolver(cls, tag_to_remove):
 _NoDatesSafeLoader.remove_implicit_resolver("tag:yaml.org,2002:timestamp")
 
 
+class SafeLineLoader(_NoDatesSafeLoader):
+    """
+    Map line number to yaml nodes.
+    https://stackoverflow.com/questions/13319067/parsing-yaml-return-with-line-number
+    """
+
+    def construct_mapping(self, node, deep=False):
+        mapping = super(SafeLineLoader, self).construct_mapping(
+            node, deep=deep)
+        for value in node.value:
+            if value[0].value != "$schema":
+                # pings
+                if node.start_mark.column == 2 and value[0].value == "description":
+                    mapping['defined_in'] = str(node.start_mark.line)
+                # metrics
+                if node.start_mark.column == 4 and value[0].value == "type":
+                    mapping['defined_in'] = str(node.start_mark.line)
```

Comment:
I think I understand why this is so hard now, thanks to your response ;)  It seems a bit brittle (as in, what if we change the names of the fields in future), but hopefully some good unit tests will help us catch that.

Can you maybe add a comment to this function about why this approach is used (kind of like your comment to my question in the PR)?  Also, let's note that the column numbers work because we run `yamllint` on the input that enforces 2-space tab width -- without that, YAML writers could use whatever tab width they want and this would break.